rotkehlchen/data_import/importers/rotki_events.py

```diff
@@ -0,0 +1,122 @@
+import csv
+from pathlib import Path
+from typing import Any, Dict, List
+
+from rotkehlchen.accounting.structures.balance import Balance
+from rotkehlchen.accounting.structures.base import HistoryBaseEntry
+from rotkehlchen.accounting.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.assets.utils import symbol_to_asset_or_token
+from rotkehlchen.constants import ZERO
+from rotkehlchen.data_import.utils import BaseExchangeImporter, UnsupportedCSVEntry
+from rotkehlchen.db.drivers.gevent import DBCursor
+from rotkehlchen.errors.asset import UnknownAsset
+from rotkehlchen.errors.misc import InputError
+from rotkehlchen.errors.serialization import DeserializationError
+from rotkehlchen.serialization.deserialize import (
+    deserialize_asset_amount,
+    deserialize_timestamp_from_date,
+)
+from rotkehlchen.types import Fee, Location, TimestampMS
+
+GENERIC_TYPE_TO_HISTORY_EVENT_TYPE_MAPPINGS = {
+    'Deposit': (HistoryEventType.DEPOSIT, HistoryEventSubType.SPEND),
+    'Withdrawal': (HistoryEventType.WITHDRAWAL, HistoryEventSubType.RECEIVE),
+    'Income': (HistoryEventType.RECEIVE, HistoryEventSubType.RECEIVE),
+    'Loss': (HistoryEventType.SPEND, HistoryEventSubType.NONE),
+    'Staking': (HistoryEventType.STAKING, HistoryEventSubType.REWARD),
+}
+
+
+class RotkiGenericEventsImporter(BaseExchangeImporter):
+    def _consume_rotki_events(
+            self,
+            cursor: DBCursor,
+            csv_row: Dict[str, Any],
+            timestamp_format: str = '%Y-%m-%d %H:%M:%S',
+    ) -> None:
+        """Consume Rotki generic events import CSV file.
+        May raise:
+        - UnsupportedCSVEntry if an unknown type is encountered.
+        - DeserializationError
+        - UnknownAsset
+        - KeyError
+        """
+        try:
+            event_type, event_subtype = GENERIC_TYPE_TO_HISTORY_EVENT_TYPE_MAPPINGS[csv_row['Type']]  # noqa: E501
+        except KeyError as e:
+            raise UnsupportedCSVEntry(f'Unsupported entry {csv_row["Type"]}. Data: {csv_row}') from e  # noqa: E501
```

Comment:
Add the identifier of the row to detect it faster

Reply:
Since the entire row is returned here, is that still needed?